app/code/Magento/InventoryBundleIndexer/Indexer/SourceItem/BundleIndexDataProvider.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryBundleIndexer\Indexer\SourceItem;
+
+use ArrayIterator;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Inventory\Model\ResourceModel\SourceItem;
+use Magento\InventoryIndexer\Indexer\IndexStructure;
+use Magento\InventoryIndexer\Model\StockIndexTableNameResolver;
+
+/**
+ * Prepare index data for bundle product.
+ */
+class BundleIndexDataProvider
+{
+    /**
+     * @var StockIndexTableNameResolver
+     */
+    private $stockIndexTableNameResolver;
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @param ResourceConnection $resourceConnection
+     * @param StockIndexTableNameResolver $stockIndexTableNameResolver
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        StockIndexTableNameResolver $stockIndexTableNameResolver
+    ) {
+        $this->stockIndexTableNameResolver = $stockIndexTableNameResolver;
+        $this->resourceConnection = $resourceConnection;
+    }
+
+    /**
+     * @param array $bundleChildrenSourceItemsSkus
+     * @param int $stockId
+     * @param string $bundleSku
+     * @return ArrayIterator
+     */
+    public function execute(array $bundleChildrenSourceItemsSkus, int $stockId, string $bundleSku): ArrayIterator
```

Comment:
Looks like we provide `$bundleSku` only for return it

Reply:
Fixed.